### DISCUSSION TITLE
Is specific http query parameters for segment URIs and key URIs in the case of HLS streams, on scope?

### DISCUSSION THREAD

**Discussion by fazilfazfaz** [streamlink/streamlink#5992] [Q&A] [answered]
### Description

There are some services that use custom query parameters for segments and keys, for example `sproutvideo`.
The existing `--http-query-param` option binds the same query parameters to all requests.
I had a working implementation and was wondering if something like this is in scope for the project.
Or would this be something relegated to individual plugins.

**Comment by bastimeyer** [comment, marked as answer]
If any required query string parameters are not part of the URLs listed in an HLS playlist (media playlists, segments or encryption keys), then this is off-spec behavior by the server. In this case, it requires a custom Streamlink plugin implementation with subclasses of Streamlink's HLS implementation, so that these specific request types can be overridden with custom logic. You can see this in some plugins where some kind of token-refresh logic is implemented for example.

**Reply by fazilfazfaz** [reply]
Ya it is off-spec and can only remember sprout using it 
Will look in to turning this in to a plugin then
Thanks

**Comment by fazilfazfaz** [comment]
> If any required query string parameters are not part of the URLs listed in an HLS playlist (media playlists, segments or encryption keys), then this is off-spec behavior by the server. In this case, it requires a custom Streamlink plugin implementation with subclasses of Streamlink's HLS implementation, so that these specific request types can be overridden with custom logic. You can see this in some plugins where some kind of token-refresh logic is implemented for example.
